src/Fabulous.Avalonia/Views/Controls/AutoCompleteBox.fs

```diff
@@ -80,6 +80,24 @@ module AutoCompleteBox =
 
                     target.Loaded.AddHandler(bindAndCleanUp))
 
+    /// Allows setting the ItemTemplate on an AutoCompleteBox
+    let ItemTemplate =
+        Attributes.defineSimpleScalar<obj -> Widget>
+            "AutoCompleteBox_ItemTemplate"
+            (fun a b ->
+                if LanguagePrimitives.PhysicalEquality a b then
+                    ScalarAttributeComparison.Identical
+                else
+                    ScalarAttributeComparison.Different)
```

Comment:
Is this implementation correct? I'm really just making stuff up that compiles here - because I don't know whether I fully understand what this equality comparison is about. Consider that the the values `a` and `b` compared here are templates compiled on-the-fly using `WidgetHelpers.compileTemplate` in the `itemTemplate` extension below. Can I use `LanguagePrimitives.PhysicalEquality` on them?

Reply:
For the first part f the question see my previous comment.

Regarding `LanguagePrimitives.PhysicalEquality`. I think is correct and arguably should be used as a default comparison on `ScalarAttributeComparers.equalityCompare` as it seems to be more safe base on this [explanation](https://stackoverflow.com/questions/39217116/how-do-i-check-for-reference-equality-in-f). cc @TimLariviere

Reply:
Moved LanguagePrimitives.PhysicalEquality to it's own module so we can reuse it.